Issues with qpython_priv.cpp during Build

**Section**
   506     // Initialize sys.argv (https://github.com/thp/pyotherside/issues/77)
   507     int argc = 1;
   508     wchar_t *argv[argc];
   509      argv[0] = Py_DecodeLocale("", nullptr);
   510      PySys_SetArgvEx(argc, argv, 0);
   511      PyMem_RawFree((void *)argv[0]);

**Issues** (Qt 5.2.1, MSVC2010 & SDK 7)
Line 508: Expected constant Expression 
Line 508: Cannot allocate an array of constant size 0
Line 508: 'argv' : unknown size
Line 509: 'Py_DecodeLocal': identifier not found
Line 511: 'PyMem_RawFree': identifier not found

Well, there are the issues I get during the build process on Windows 7. 

What should I do? My C++ Knowledge is slim.

qpython_priv.cpp(522): error C2131: expression did not evaluate to a constant
qpython_priv.cpp(522): note: failure was caused by a read of a variable outside its lifetime
qpython_priv.cpp(522): note: see usage of 'argc'
qpython_priv.cpp(523): error C3861: 'Py_DecodeLocale': identifier not found
qpython_priv.cpp(599): error C2664: 'PyObject *PyObject_CallMethod(PyObject *,char *,char *,...)': cannot convert argument 2 from 'const char [17]' to 'char *'
qpython_priv.cpp(600): note: Conversion from string literal loses const qualifier (see /Zc:strictStrings)
jom: D:\pyotherside\src\Makefile.Release [release\qpython_priv.obj] Error 2
	cl -c -nologo -Zc:wchar_t -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -Zc:referenceBinding -Zc:__cplusplus -IC:\Python33\include -O2 -MD -W3 -w34100 -w34189 -w44996 -w44456 -w44457 -w44458 -wd4577 -wd4467 -EHsc -DUNICODE -D_UNICODE -DWIN32 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -DPYOTHERSIDE_VERSION=\"1.5.4\" -DQT_NO_DEBUG -DQT_PLUGIN -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DNDEBUG -D_WINDLL -I. -I. -IC:\Qt\5.12.3\msvc2017\include -IC:\Qt\5.12.3\msvc2017\include\QtQuick -IC:\Qt\5.12.3\msvc2017\include\QtSvg -IC:\Qt\5.12.3\msvc2017\include\QtWidgets -IC:\Qt\5.12.3\msvc2017\include\QtGui -IC:\Qt\5.12.3\msvc2017\include\QtANGLE -IC:\Qt\5.12.3\msvc2017\include\QtQml -IC:\Qt\5.12.3\msvc2017\include\QtNetwork -IC:\Qt\5.12.3\msvc2017\include\QtCore -Irelease -IC:\VulkanSDK\1.0.51.0\include -IC:\Qt\5.12.3\msvc2017\mkspecs\win32-msvc -Forelease\ @C:\Users\jzlotnik\AppData\Local\Temp\moc_qobject_ref.obj.10680.766.jom
moc_qobject_ref.cpp
	cl -c -nologo -Zc:wchar_t -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -Zc:referenceBinding -Zc:__cplusplus -IC:\Python33\include -O2 -MD -W3 -w34100 -w34189 -w44996 -w44456 -w44457 -w44458 -wd4577 -wd4467 -EHsc -DUNICODE -D_UNICODE -DWIN32 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -DPYOTHERSIDE_VERSION=\"1.5.4\" -DQT_NO_DEBUG -DQT_PLUGIN -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DNDEBUG -D_WINDLL -I. -I. -IC:\Qt\5.12.3\msvc2017\include -IC:\Qt\5.12.3\msvc2017\include\QtQuick -IC:\Qt\5.12.3\msvc2017\include\QtSvg -IC:\Qt\5.12.3\msvc2017\include\QtWidgets -IC:\Qt\5.12.3\msvc2017\include\QtGui -IC:\Qt\5.12.3\msvc2017\include\QtANGLE -IC:\Qt\5.12.3\msvc2017\include\QtQml -IC:\Qt\5.12.3\msvc2017\include\QtNetwork -IC:\Qt\5.12.3\msvc2017\include\QtCore -Irelease -IC:\VulkanSDK\1.0.51.0\include -IC:\Qt\5.12.3\msvc2017\mkspecs\win32-msvc -Forelease\ @C:\Users\jzlotnik\AppData\Local\Temp\moc_qpython_priv.obj.10680.812.jom
moc_qpython_priv.cpp
c:\python33\include\pymath.h(22): warning C4273: 'round': inconsistent dll linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt_math.h(533): note: see previous definition of 'round'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include\xtgmath.h(221): warning C4273: 'round': inconsistent dll linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt_math.h(533): note: see previous definition of 'round'
	cl -c -nologo -Zc:wchar_t -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -Zc:referenceBinding -Zc:__cplusplus -IC:\Python33\include -O2 -MD -W3 -w34100 -w34189 -w44996 -w44456 -w44457 -w44458 -wd4577 -wd4467 -EHsc -DUNICODE -D_UNICODE -DWIN32 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -DPYOTHERSIDE_VERSION=\"1.5.4\" -DQT_NO_DEBUG -DQT_PLUGIN -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DNDEBUG -D_WINDLL -I. -I. -IC:\Qt\5.12.3\msvc2017\include -IC:\Qt\5.12.3\msvc2017\include\QtQuick -IC:\Qt\5.12.3\msvc2017\include\QtSvg -IC:\Qt\5.12.3\msvc2017\include\QtWidgets -IC:\Qt\5.12.3\msvc2017\include\QtGui -IC:\Qt\5.12.3\msvc2017\include\QtANGLE -IC:\Qt\5.12.3\msvc2017\include\QtQml -IC:\Qt\5.12.3\msvc2017\include\QtNetwork -IC:\Qt\5.12.3\msvc2017\include\QtCore -Irelease -IC:\VulkanSDK\1.0.51.0\include -IC:\Qt\5.12.3\msvc2017\mkspecs\win32-msvc -Forelease\ @C:\Users\jzlotnik\AppData\Local\Temp\moc_pyotherside_plugin.obj.10680.1094.jom
moc_pyotherside_plugin.cpp
	cl -c -nologo -Zc:wchar_t -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -Zc:referenceBinding -Zc:__cplusplus -IC:\Python33\include -O2 -MD -W3 -w34100 -w34189 -w44996 -w44456 -w44457 -w44458 -wd4577 -wd4467 -EHsc -DUNICODE -D_UNICODE -DWIN32 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -DPYOTHERSIDE_VERSION=\"1.5.4\" -DQT_NO_DEBUG -DQT_PLUGIN -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DNDEBUG -D_WINDLL -I. -I. -IC:\Qt\5.12.3\msvc2017\include -IC:\Qt\5.12.3\msvc2017\include\QtQuick -IC:\Qt\5.12.3\msvc2017\include\QtSvg -IC:\Qt\5.12.3\msvc2017\include\QtWidgets -IC:\Qt\5.12.3\msvc2017\include\QtGui -IC:\Qt\5.12.3\msvc2017\include\QtANGLE -IC:\Qt\5.12.3\msvc2017\include\QtQml -IC:\Qt\5.12.3\msvc2017\include\QtNetwork -IC:\Qt\5.12.3\msvc2017\include\QtCore -Irelease -IC:\VulkanSDK\1.0.51.0\include -IC:\Qt\5.12.3\msvc2017\mkspecs\win32-msvc -Forelease\ @C:\Users\jzlotnik\AppData\Local\Temp\moc_pyglarea.obj.10680.844.jom
moc_pyglarea.cpp
c:\python33\include\pymath.h(22): warning C4273: 'round': inconsistent dll linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt_math.h(533): note: see previous definition of 'round'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include\xtgmath.h(221): warning C4273: 'round': inconsistent dll linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt_math.h(533): note: see previous definition of 'round'
	cl -c -nologo -Zc:wchar_t -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -Zc:referenceBinding -Zc:__cplusplus -IC:\Python33\include -O2 -MD -W3 -w34100 -w34189 -w44996 -w44456 -w44457 -w44458 -wd4577 -wd4467 -EHsc -DUNICODE -D_UNICODE -DWIN32 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -DPYOTHERSIDE_VERSION=\"1.5.4\" -DQT_NO_DEBUG -DQT_PLUGIN -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DNDEBUG -D_WINDLL -I. -I. -IC:\Qt\5.12.3\msvc2017\include -IC:\Qt\5.12.3\msvc2017\include\QtQuick -IC:\Qt\5.12.3\msvc2017\include\QtSvg -IC:\Qt\5.12.3\msvc2017\include\QtWidgets -IC:\Qt\5.12.3\msvc2017\include\QtGui -IC:\Qt\5.12.3\msvc2017\include\QtANGLE -IC:\Qt\5.12.3\msvc2017\include\QtQml -IC:\Qt\5.12.3\msvc2017\include\QtNetwork -IC:\Qt\5.12.3\msvc2017\include\QtCore -Irelease -IC:\VulkanSDK\1.0.51.0\include -IC:\Qt\5.12.3\msvc2017\mkspecs\win32-msvc -Forelease\ @C:\Users\jzlotnik\AppData\Local\Temp\moc_pyfbo.obj.10680.859.jom
moc_pyfbo.cpp
c:\python33\include\pymath.h(22): warning C4273: 'round': inconsistent dll linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt_math.h(533): note: see previous definition of 'round'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include\xtgmath.h(221): warning C4273: 'round': inconsistent dll linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt_math.h(533): note: see previous definition of 'round'
	cl -c -nologo -Zc:wchar_t -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -Zc:referenceBinding -Zc:__cplusplus -IC:\Python33\include -O2 -MD -W3 -w34100 -w34189 -w44996 -w44456 -w44457 -w44458 -wd4577 -wd4467 -EHsc -DUNICODE -D_UNICODE -DWIN32 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -DPYOTHERSIDE_VERSION=\"1.5.4\" -DQT_NO_DEBUG -DQT_PLUGIN -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DNDEBUG -D_WINDLL -I. -I. -IC:\Qt\5.12.3\msvc2017\include -IC:\Qt\5.12.3\msvc2017\include\QtQuick -IC:\Qt\5.12.3\msvc2017\include\QtSvg -IC:\Qt\5.12.3\msvc2017\include\QtWidgets -IC:\Qt\5.12.3\msvc2017\include\QtGui -IC:\Qt\5.12.3\msvc2017\include\QtANGLE -IC:\Qt\5.12.3\msvc2017\include\QtQml -IC:\Qt\5.12.3\msvc2017\include\QtNetwork -IC:\Qt\5.12.3\msvc2017\include\QtCore -Irelease -IC:\VulkanSDK\1.0.51.0\include -IC:\Qt\5.12.3\msvc2017\mkspecs\win32-msvc -Forelease\ @C:\Users\jzlotnik\AppData\Local\Temp\moc_qpython.obj.10680.937.jom
moc_qpython.cpp
c:\python33\include\pymath.h(22): warning C4273: 'round': inconsistent dll linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt_math.h(533): note: see previous definition of 'round'
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include\xtgmath.h(221): warning C4273: 'round': inconsistent dll linkage
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt\corecrt_math.h(533): note: see previous definition of 'round'
	cl -c -nologo -Zc:wchar_t -FS -Zc:rvalueCast -Zc:inline -Zc:strictStrings -Zc:throwingNew -Zc:referenceBinding -Zc:__cplusplus -IC:\Python33\include -O2 -MD -W3 -w34100 -w34189 -w44996 -w44456 -w44457 -w44458 -wd4577 -wd4467 -EHsc -DUNICODE -D_UNICODE -DWIN32 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -DPYOTHERSIDE_VERSION=\"1.5.4\" -DQT_NO_DEBUG -DQT_PLUGIN -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -DNDEBUG -D_WINDLL -I. -I. -IC:\Qt\5.12.3\msvc2017\include -IC:\Qt\5.12.3\msvc2017\include\QtQuick -IC:\Qt\5.12.3\msvc2017\include\QtSvg -IC:\Qt\5.12.3\msvc2017\include\QtWidgets -IC:\Qt\5.12.3\msvc2017\include\QtGui -IC:\Qt\5.12.3\msvc2017\include\QtANGLE -IC:\Qt\5.12.3\msvc2017\include\QtQml -IC:\Qt\5.12.3\msvc2017\include\QtNetwork -IC:\Qt\5.12.3\msvc2017\include\QtCore -Irelease -IC:\VulkanSDK\1.0.51.0\include -IC:\Qt\5.12.3\msvc2017\mkspecs\win32-msvc -Forelease\ @C:\Users\jzlotnik\AppData\Local\Temp\moc_qpython_worker.obj.10680.781.jom
moc_qpython_worker.cpp
jom: D:\pyotherside\src\Makefile [release] Error 2
jom: D:\pyotherside\Makefile [sub-src-make_first] Error 2
00:21:45: The process "C:\Qt\Tools\QtCreator\bin\jom.exe" exited with code 2.
Error while building/deploying project pyotherside (kit: Desktop Qt 5.12.3 MSVC2017 32bit)
The kit Desktop Qt 5.12.3 MSVC2017 32bit has configuration issues which might be the root cause for this problem.
When executing step "Make"
00:21:45: Elapsed time: 00:06.

--- a/src/qpython_priv.cpp
+++ b/src/qpython_priv.cpp
@@ -519,10 +519,11 @@ QPythonPriv::QPythonPriv()
 
     // Initialize sys.argv (https://github.com/thp/pyotherside/issues/77)
     int argc = 1;
-    wchar_t *argv[argc];
+    wchar_t **argv = (wchar_t **)malloc(argc * sizeof(wchar_t *));
     argv[0] = Py_DecodeLocale("", nullptr);
     PySys_SetArgvEx(argc, argv, 0);
     PyMem_RawFree((void *)argv[0]);
+    free(argv);
 
     locals = PyObjectRef(PyDict_New(), true);
     assert(locals);

